src/btree/bt_read.c

```diff
@@ -257,6 +257,11 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
             if (!LF_ISSET(WT_READ_IGNORE_CACHE_SIZE))
                 WT_RET(__wt_cache_eviction_check(
                   session, true, !F_ISSET(session->txn, WT_TXN_HAS_ID), NULL));
+
+            /* Rollback the transaction if the operation timer fires. */
+            if (__wt_op_timer_fired(session))
```

Comment:
Need to check the timer here again as we are not allowed to return `WT_ROLLBACK` from `__wt_cache_eviction_check`

Reply:
Could you explain the code flow that leads to us wanting to `WT_ROLLBACK` in `__wt_page_in_func` it seems like an odd place to do it. I know we can't return it from the other location, which is why you moved it here but I don't understand why we're rolling back in the first place.

Reply:
An old reader is pinning all the internal pages which are newly split after the reader enter the tree in cache. The operation timeout mechanism should rollback this reader to free the internal pages. However, that mechanism is not working because it assumes there should not be a history store cursor open when the timer starts, which is not true. I fixed this issue but now the problem becomes `__wt_page_in_func ` will try to load the page again and again and we are still stuck. This is because `__wt_cache_eviction_check ` returns 0 even the timer has fired so there is no cache being freed and the following `page_read` call will fail because there is not enough space. We will then retry this infinitely. Since we can't return WT_ROLLBACK directly from `__wt_cache_eviction_check `, so I have to check the timer again here to break the loop.